frontend/src/pages/course/components/LeaveCourse/CourseMembershipService.tsx

```diff
@@ -0,0 +1,54 @@
+import apiCall from "../../../../util/apiFetch";
+import { ApiRoutes } from "../../../../@types/requests.d";
+import { TFunction } from "i18next";
+import axios, {AxiosError} from "axios";
+
+
+export interface LeaveStatus {
+    success: boolean;
+    message: string;
+}
+
+export const leaveCourse = async (courseId: string, t: TFunction<"translation", undefined> | undefined): Promise<LeaveStatus> => {
+    try {
+        const response = await apiCall.delete(ApiRoutes.COURSE_LEAVE, undefined, { courseId: courseId });
+
+        // Check if the response status is OK (200)
+        if (response.status === 200) {
+            return {
+                success: true,
+                message: t ? t("course.leaveSuccess") : "Left course successfully"
+            };
+        } else {
+            // Extract error message from response data
+            const errorMessage = response.data?.message || "Failed to leave the course"
+            return {
```

Comment:
Alle 2xx statussen worden als success beschouwd. Dus deze error mag niet getoond worden. Als je een 4xx of 5xx error krijgt dan gooit axios een error.

Reply:
Ok, bedankt voor de verduidelijking!